File: flutter_application_1/api/bien.php
<?php
header('Content-Type: application/json; charset=utf-8');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: GET, POST, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');

$host = 'localhost';
$user = 'root';
$password = '';
$database = 'holidaze';

try {
    if (!isset($_GET['id']) || empty($_GET['id'])) {
        throw new Exception('ID du bien requis');
    }
    
    $id_bien = (int)$_GET['id'];
    
    $conn = new mysqli($host, $user, $password, $database);
    
    if ($conn->connect_error) {
        throw new Exception('Erreur de connexion: ' . $conn->connect_error);
    }
    
    $conn->set_charset('utf8mb4');
    
    // Récupérer les détails du bien
    $sql = "
        SELECT 
            b.id_bien,
            b.nom_bien,
            b.rue_bien,
            b.description_bien,
            b.superficie_bien,
            b.animaux_bien,
            b.nb_couchage,
            b.latitude_bien,
            b.longitude_bien,
            c.nom_commune,
            c.id_commune,
            COALESCE(ROUND(AVG(a.note), 1), 0) as note_moyenne,
            COUNT(DISTINCT a.id_avis) as nb_avis,
            150 as prix_nuit
        FROM bien b
        LEFT JOIN commune c ON b.id_commune = c.id_commune
        LEFT JOIN avis a ON b.id_bien = a.id_bien AND a.valide = 1
        WHERE b.id_bien = ?
        GROUP BY b.id_bien
    ";
    
    $stmt = $conn->prepare($sql);
    if (!$stmt) {
        throw new Exception('Erreur de préparation: ' . $conn->error);
    }
    
    $stmt->bind_param("i", $id_bien);
    $stmt->execute();
    $result = $stmt->get_result();
    
    if ($result->num_rows === 0) {
        throw new Exception('Bien non trouvé');
    }
    
    $row = $result->fetch_assoc();
    
    // Récupérer les prestations du bien
    $prestSql = "
        SELECT p.id_prestation, p.libelle_prestation
        FROM secompose sc
        JOIN prestation p ON sc.id_prestation = p.id_prestation
        WHERE sc.id_bien = ?
    ";
    
    $prestStmt = $conn->prepare($prestSql);
    $prestStmt->bind_param("i", $id_bien);
    $prestStmt->execute();
    $prestResult = $prestStmt->get_result();
    
    $prestations = [];
    while ($pRow = $prestResult->fetch_assoc()) {
        $prestations[] = [
            'id' => (int)$pRow['id_prestation'],
            'nom' => $pRow['libelle_prestation']
        ];
    }
    
    // Récupérer les avis validés
    $avisSql = "
        SELECT 
            a.id_avis,
            a.note,
            a.commentaire,
            a.date_avis
        FROM avis a
        WHERE a.id_bien = ? AND a.valide = 1
        ORDER BY a.date_avis DESC
        LIMIT 10
    ";
    
    $avisStmt = $conn->prepare($avisSql);
    $avisStmt->bind_param("i", $id_bien);
    $avisStmt->execute();
    $avisResult = $avisStmt->get_result();
    
    $avis = [];
    while ($aRow = $avisResult->fetch_assoc()) {
        $avis[] = [
            'id' => (int)$aRow['id_avis'],
            'note' => (int)$aRow['note'],
            'commentaire' => $aRow['commentaire'],
            'date' => $aRow['date_avis']
        ];
    }
    
    $stmt->close();
    $prestStmt->close();
    $avisStmt->close();
    $conn->close();
    
    // Générer une URL d'image placeholder unique
    $imageId = 400 + $id_bien;
    $photoUrl = 'https://picsum.photos/800/600?random=' . $imageId;
    
    echo json_encode([
        'success' => true,
        'data' => [
            'id_bien' => (int)$row['id_bien'],
            'nom_bien' => $row['nom_bien'],
            'rue_bien' => $row['rue_bien'],
            'description_bien' => $row['description_bien'],
            'superficie_bien' => $row['superficie_bien'],
            'animaux_bien' => $row['animaux_bien'],
            'nb_couchage' => $row['nb_couchage'],
            'ville' => $row['nom_commune'] ?? 'Non spécifiée',
            'latitude' => (float)$row['latitude_bien'],
            'longitude' => (float)$row['longitude_bien'],
            'prix_nuit' => (int)$row['prix_nuit'],
            'note_moyenne' => (float)$row['note_moyenne'],
            'nb_avis' => (int)$row['nb_avis'],
            'photo_url' => $photoUrl,
            'prestations' => $prestations,
            'avis' => $avis
        ]
    ]);
    
} catch (Exception $e) {
    http_response_code(500);
    echo json_encode([
        'success' => false,
        'error' => $e->getMessage()
    ]);
}
?>

File: flutter_application_1/lib/models/bien.dart
class Bien {
  final int idBien;
  final String nomBien;
  final String rueBien;
  final String descriptionBien;
  final String superficieBien;
  final String animauxBien;
  final int nbCouchage;
  final String ville;
  final double latitude;
  final double longitude;
  final int prixNuit;
  final double noteMoyenne;
  final int nbAvis;
  final String photoUrl;
  final List<Prestation>? prestations;
  final List<Avis>? avis;

  Bien({
    required this.idBien,
    required this.nomBien,
    required this.rueBien,
    required this.descriptionBien,
    required this.superficieBien,
    required this.animauxBien,
    required this.nbCouchage,
    required this.ville,
    required this.latitude,
    required this.longitude,
    required this.prixNuit,
    required this.noteMoyenne,
    required this.nbAvis,
    required this.photoUrl,
    this.prestations,
    this.avis,
  });

  factory Bien.fromJson(Map<String, dynamic> json) {
    int parseIntValue(dynamic value) {
      if (value == null) return 0;
      if (value is int) return value;
      if (value is double) return value.toInt();
      if (value is String) return int.tryParse(value) ?? 0;
      return 0;
    }

    double parseDoubleValue(dynamic value) {
      if (value == null) return 0.0;
      if (value is double) return value;
      if (value is int) return value.toDouble();
      if (value is String) return double.tryParse(value) ?? 0.0;
      return 0.0;
    }

    return Bien(
      idBien: parseIntValue(json['id_bien']),
      nomBien: (json['nom_bien'] as String?)?.trim() ?? '',
      rueBien: (json['rue_bien'] as String?)?.trim() ?? '',
      descriptionBien: (json['description_bien'] as String?)?.trim() ?? '',
      superficieBien: (json['superficie_bien'] as String?)?.trim() ?? '',
      animauxBien: (json['animaux_bien'] as String?)?.trim() ?? 'Non',
      nbCouchage: parseIntValue(json['nb_couchage']),
      ville: (json['ville'] as String?)?.trim() ?? '',
      latitude: parseDoubleValue(json['latitude']),
      longitude: parseDoubleValue(json['longitude']),
      prixNuit: parseIntValue(json['prix_nuit']),
      noteMoyenne: parseDoubleValue(json['note_moyenne']),
      nbAvis: parseIntValue(json['nb_avis']),
      photoUrl: (json['photo_url'] as String?)?.trim() ?? '',
      prestations:
          json['prestations'] != null
              ? (json['prestations'] as List)
                  .map((p) => Prestation.fromJson(p as Map<String, dynamic>))
                  .toList()
              : null,
      avis:
          json['avis'] != null
              ? (json['avis'] as List)
                  .map((a) => Avis.fromJson(a as Map<String, dynamic>))
                  .toList()
              : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_bien': idBien,
      'nom_bien': nomBien,
      'rue_bien': rueBien,
      'description_bien': descriptionBien,
      'superficie_bien': superficieBien,
      'animaux_bien': animauxBien,
      'nb_couchage': nbCouchage,
      'ville': ville,
      'latitude': latitude,
      'longitude': longitude,
      'prix_nuit': prixNuit,
      'note_moyenne': noteMoyenne,
      'nb_avis': nbAvis,
      'photo_url': photoUrl,
      'prestations': prestations?.map((p) => p.toJson()).toList(),
      'avis': avis?.map((a) => a.toJson()).toList(),
    };
  }

  @override
  String toString() =>
      'Bien(id: $idBien, nom: $nomBien, ville: $ville, prix: $prixNuit€, note: $noteMoyenne/5)';
}

class Prestation {
  final int id;
  final String nom;

  Prestation({required this.id, required this.nom});

  factory Prestation.fromJson(Map<String, dynamic> json) {
    return Prestation(
      id: json['id'] as int? ?? 0,
      nom: json['nom'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'nom': nom};
  }
}

class Avis {
  final int id;
  final int note;
  final String commentaire;
  final String date;

  Avis({
    required this.id,
    required this.note,
    required this.commentaire,
    required this.date,
  });

  factory Avis.fromJson(Map<String, dynamic> json) {
    return Avis(
      id: json['id'] as int? ?? 0,
      note: json['note'] as int? ?? 0,
      commentaire: json['commentaire'] as String? ?? '',
      date: json['date'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'note': note, 'commentaire': commentaire, 'date': date};
  }
}
